src/QAT/qat/purr/backends/qiskit_simulator.py

```diff
@@ -255,7 +255,7 @@ def execute(self, builder: QiskitBuilder) -> Dict[str, Any]:
 
         # Because qiskit needs all values up-front we just provide a maximal classical register then trim off
         # the values we aren't going to use.
-        trimmed = {key[removals:]: value for key, value in distribution.items()}
+        trimmed = {key[removals:][::-1]: value for key, value in distribution.items()}
```

Comment:
I believe it's worthwhile to address the issue where reversion actually comes from. I suspect the `QiskitBuilder.measure()` needs to ensure MSQub <-> MSClb (https://quantumcomputing.stackexchange.com/questions/8893/why-is-the-order-reversed-on-measurement)


```
    def measure(self, target: Qubit, *args, **kwargs) -> "InstructionBuilder":
        self.circuit.measure(target.index, self.bit_count)
        self.bit_count = self.bit_count + 1
        return self
```

Reply:
https://docs.quantum.ibm.com/guides/bit-ordering

Alas, the `reverse_bits` is only for drawing, but I believe we can tweak on our end